### DISCUSSION TITLE
Configuration File

### DISCUSSION THREAD

**Discussion by er83** [gabime/spdlog#2501] [Q&A] [answered]
We are in the process of upgrading to using version 1.10. 
The configuration.h is no longer available to allow initialization via configuration file. 
There are references to using https://github.com/guangie88/spdlog_setup using toml file.
Is this the recommended way?

**Comment by tt4g** [comment]
That project is independent of spdlog. If you have a problem, please open an issue in that project.

**Comment by er83** [comment]
Is there a way to initialize via configuration file in spdlog? 
The configuration.h file that declares cspdlog::configuration::create is no longer available in version 1.10 of spdlog.
Thanks

**Reply by tt4g** [reply]
> Is there a way to initialize via configuration file in spdlog?

Yes.

What is `configuration.h`?

**Comment by er83** [comment, marked as answer]
The header configuration.h was available in the older versions of spdlog. This file is no longer available in version 1.10
Previously to initialise using config file, you would call method 'spdlog::configuration::create' which was available in configuration.h .
Can you please suggest how you would initialise using config file in version 1.10 ?

**Reply by tt4g** [reply]
Which version of spdlog?
The oldest tag v0.9.0 does not seem to have it.

**Reply by er83** [reply]
Should be Version 0.16.3. Strange can't see it there.

Could you advise as to how I can initialise using config file in version 1.10?

Thanks

**Reply by tt4g** [reply]
As far as I know spdlog has never supported the feature to initialize by configuration file.
It may be that someone else added `configuration.h` to the spdlog source code that you downloaded, but we don't know the truth.

Try using the https://github.com/guangie88/spdlog_setup library you found.

**Reply by er83** [reply]
Thanks.
It was source downloaded before my time. Perhaps that was the case.
Thanks for confirming that there has never been support for initialization by configuration file

**Comment by gabime** [comment]
> As far as I know spdlog has never supported the feature to initialize by configuration file.

Can confirm. It was never part of spdlog.
